## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-fs.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-xl.sh | :x: [more...](#3b70e2a0-a669-46fd-be13-c8f4a487ad1d) |
| mint-large-bucket.sh | :x: [more...](#a4f12541-aa44-404b-89a4-e4d96a7ab93a) |
| mint-dist-xl.sh | :x: [more...](#5a3ffdf7-6c89-4edd-b4f9-c3a44c49df72) |
| mint-gateway-azure.sh | :x: [more...](#209191ce-d006-4d05-b626-aa9986e19648) |
<a name="209191ce-d006-4d05-b626-aa9986e19648"></a>
### 9917-3821458/mint-gateway-azure.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-dev7.minio.io:31728
ACCESS_KEY:           minioazure
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 8efe6b288071:/mint/log /tmp/mint-logs'

(1/15) Running aws-sdk-go tests ... done in 8 seconds
(2/15) Running aws-sdk-java tests ... done in 1 seconds
(3/15) Running aws-sdk-php tests ... done in 2 minutes and 9 seconds
(4/15) Running aws-sdk-ruby tests ... done in 16 seconds
(5/15) Running awscli tests ... done in 4 minutes and 5 seconds
(6/15) Running healthcheck tests ... done in 0 seconds
(7/15) Running mc tests ... done in 3 minutes and 5 seconds
(8/15) Running minio-dotnet tests ... done in 2 minutes and 33 seconds
(9/15) Running minio-go tests ... done in 5 minutes and 11 seconds
(10/15) Running minio-java tests ... FAILED in 27 seconds
{
  "name": "minio-java",
  "function": "getObject(String bucketName, String objectName, long offset, Long length)",
  "args": "offset: 1000, length: 64 KB",
  "duration": 1304,
  "status": "FAIL",
  "error": "java.io.EOFException >>> [okio.RealBufferedSink.write(RealBufferedSink.java:115), io.minio.HttpRequestBody.writeTo(HttpRequestBody.java:78), okhttp3.internal.http.CallServerInterceptor.intercept(CallServerInterceptor.java:83), okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:147), okhttp3.internal.connection.ConnectInterceptor.intercept(ConnectInterceptor.java:45), okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:147), okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:121), okhttp3.internal.cache.CacheInterceptor.intercept(CacheInterceptor.java:94), okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:147), okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:121), okhttp3.internal.http.BridgeInterceptor.intercept(BridgeInterceptor.java:93), okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:147), okhttp3.internal.http.RetryAndFollowUpInterceptor.intercept(RetryAndFollowUpInterceptor.java:125), okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:147), okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:121), okhttp3.RealCall.getResponseWithInterceptorChain(RealCall.java:264), okhttp3.RealCall.execute(RealCall.java:93), io.minio.MinioClient.execute(MinioClient.java:1097), io.minio.MinioClient.execute(MinioClient.java:1236), io.minio.MinioClient.executePut(MinioClient.java:1410), io.minio.MinioClient.executePut(MinioClient.java:1424), io.minio.MinioClient.uploadPart(MinioClient.java:5492), io.minio.MinioClient.putObject(MinioClient.java:3740), io.minio.MinioClient.putObject(MinioClient.java:3870), FunctionalTest.getObject_test3(FunctionalTest.java:1157), FunctionalTest.runTests(FunctionalTest.java:4035), FunctionalTest.main(FunctionalTest.java:4192)]"
}
(10/15) Running minio-js tests ... done in 2 minutes and 35 seconds
(11/15) Running minio-py tests ... done in 9 minutes and 37 seconds
(12/15) Running s3cmd tests ... done in 1 minutes and 21 seconds
(13/15) Running s3select tests ... done in 1 minutes and 2 seconds
(14/15) Running security tests ... done in 0 seconds

Executed 14 out of 15 tests successfully.
```
<a name="a4f12541-aa44-404b-89a4-e4d96a7ab93a"></a>
### 9917-3821458/mint-large-bucket.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-dev4.minio.io:31984
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 1f74cb940793:/mint/log /tmp/mint-logs'

(1/15) Running aws-sdk-go tests ... FAILED in 4 minutes and 7 seconds
{
  "alert": "",
  "args": {
    "bucketName": "aws-sdk-go-test-ico2ax1zk4hc2x",
    "expiry": 60000000000,
    "objectName1": "testObject1",
    "objectName2": "testObject2"
  },
  "duration": 120889,
  "error": "BucketAlreadyOwnedByYou: Your previous request to create the named bucket succeeded and you already own it.\n\tstatus code: 409, request id: 161BDAC6AA279921, host id: ",
  "function": "testListObjects",
  "message": "AWS SDK Go CreateBucket Failed",
  "name": "aws-sdk-go",
  "status": "FAIL"
}
(1/15) Running aws-sdk-java tests ... done in 7 seconds
(2/15) Running aws-sdk-php tests ... done in 1 minutes and 17 seconds
(3/15) Running aws-sdk-ruby tests ... FAILED in 7 seconds
{
  "name": "aws-sdk-ruby",
  "function": "makeBucket(bucket_name)",
  "args": {
    "bucket_name": "aws-sdk-ruby-bucket-c61796b59f25"
  },
  "duration": 323.39,
  "error": "Your previous request to create the named bucket succeeded and you already own it.",
  "status": "FAIL"
}
(3/15) Running awscli tests ... FAILED in 17 seconds
{
  "name": "awscli",
  "duration": 8668,
  "function": "delete_bucket\n",
  "status": "FAIL",
  "error": "delete: s3://awscli-mint-test-bucket-32457/datafile-1-kBremove_bucket failed: s3://awscli-mint-test-bucket-32457 An error occurred (BucketNotEmpty) when calling the DeleteBucket operation: The bucket you tried to delete is not empty"
}
(3/15) Running healthcheck tests ... done in 0 seconds
(4/15) Running mc tests ... FAILED in 1 minutes and 35 seconds
{
  "name": "mc",
  "duration": "82936",
  "function": "test_put_object_with_storage_class",
  "status": "FAIL",
  "error": "/mint/run/core/mc/mc --config-dir /tmp/.mc-2068 --quiet --no-color cp --storage-class REDUCED_REDUNDANCY /mint/data/datafile-1-MB myminio/mc-test-bucket-16891/mc-test-object-32021  >>> \n`/mint/data/datafile-1-MB` -> `myminio/mc-test-bucket-16891/mc-test-object-32021`\nmc: <ERROR> Failed to copy `/mint/data/datafile-1-MB`. Please reduce your request\nTotal: 0 B, Transferred: 10.00 MiB, Speed: 124.26 KiB/s"
}
(4/15) Running minio-dotnet tests ... done in 3 minutes and 50 seconds
(5/15) Running minio-go tests ... FAILED in 1 seconds
{
  "args": {
    "bucketName": "minio-go-test-dugwm1om5ubcsqso",
    "region": "eu-west-1"
  },
  "duration": 922,
  "error": "Your previous request to create the named bucket succeeded and you already own it.",
  "function": "MakeBucket(bucketName, region)",
  "message": "MakeBucket failed",
  "name": "minio-go: testGetObjectClosedTwiceV2",
  "status": "FAIL"
}
(5/15) Running minio-java tests ... FAILED in 3 seconds
{
  "name": "minio-java",
  "function": "listBuckets()",
  "duration": 27,
  "status": "FAIL",
  "error": "error occurred\nErrorResponse(code = SlowDown, message = Please reduce your request, bucketName = minio-java-test-1r8g9ed, objectName = null, resource = /minio-java-test-1r8g9ed, requestId = 161BDB2C9FB3B5C6, hostId = 8c63929e-c39d-4e97-9ab1-0515ac3e40fe)\nrequest={method=PUT, url=http://minio-dev4.minio.io:31984/minio-java-test-1r8g9ed, headers=Host: minio-dev4.minio.io:31984\nAccept-Encoding: identity\nUser-Agent: MinIO (amd64; amd64) minio-java/dev\nContent-MD5: 1B2M2Y8AsgTpgAmY7PhCfg==\nx-amz-content-sha256: e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855\nx-amz-date: 20200625T175757Z\nAuthorization: AWS4-HMAC-SHA256 Credential=*REDACTED*/20200625/us-east-1/s3/aws4_request, SignedHeaders=accept-encoding;content-md5;host;x-amz-content-sha256;x-amz-date, Signature=*REDACTED*\n}\nresponse={code=503, headers=Accept-Ranges: bytes\nContent-Length: 305\nContent-Security-Policy: block-all-mixed-content\nContent-Type: application/xml\nRetry-After: 120\nServer: MinIO/DEVELOPMENT.2020-06-25T17-40-36Z\nVary: Origin\nX-Amz-Request-Id: 161BDB2C9FB3B5C6\nX-Xss-Protection: 1; mode=block\nDate: Thu, 25 Jun 2020 17:57:57 GMT\n}\n >>> [io.minio.MinioClient.execute(MinioClient.java:1211), io.minio.MinioClient.execute(MinioClient.java:1236), io.minio.MinioClient.executePut(MinioClient.java:1410), io.minio.MinioClient.makeBucket(MinioClient.java:3286), io.minio.MinioClient.makeBucket(MinioClient.java:3199), FunctionalTest.listBuckets_test(FunctionalTest.java:324), FunctionalTest.runTests(FunctionalTest.java:4011), FunctionalTest.main(FunctionalTest.java:4192)]"
}
(5/15) Running minio-js tests ... FAILED in 9 seconds
{
  "name": "minio-js",
  "function": "\"after all\" hook in \"functional tests\"",
  "duration": 9,
  "status": "FAIL",
  "error": "S3Error: Please reduce your request at Object.parseError (node_modules/minio/dist/main/xml-parsers.js:86:11) at /mint/run/core/minio-js/node_modules/minio/dist/main/transformers.js:156:22 at DestroyableTransform._flush (node_modules/minio/dist/main/transformers.js:80:10) at DestroyableTransform.prefinish (node_modules/readable-stream/lib/_stream_transform.js:129:10) at prefinish (node_modules/readable-stream/lib/_stream_writable.js:611:14) at finishMaybe (node_modules/readable-stream/lib/_stream_writable.js:620:5) at endWritable (node_modules/readable-stream/lib/_stream_writable.js:643:3) at DestroyableTransform.Writable.end (node_modules/readable-stream/lib/_stream_writable.js:571:22) at IncomingMessage.onend (_stream_readable.js:673:10) at endReadableNT (_stream_readable.js:1201:12) at processTicksAndRejections (internal/process/task_queues.js:84:21)"
}
(5/15) Running minio-py tests ... FAILED in 2 minutes and 0 seconds
{
  "name": "minio-py:test_make_bucket_default_region",
  "function": "make_bucket(bucket_name, location)",
  "args": {
    "bucket_name": "minio-py-test-1817cfdc-d7be-4c96-9b0a-b4958ebb7a06",
    "location": "default value ('us-east-1')"
  },
  "duration": 120078,
  "message": "BucketAlreadyOwnedByYou: message: Your previous request to create the named bucket succeeded and you already own it.",
  "error": "Traceback (most recent call last):\n  File \"/mint/run/core/minio-py/tests.py\", line 188, in test_make_bucket_default_region\n    client.make_bucket(bucket_name)\n  File \"/usr/local/lib/python3.6/dist-packages/minio/api.py\", line 374, in make_bucket\n    raise ResponseError(response, method, bucket_name).get_exception()\nminio.error.BucketAlreadyOwnedByYou: BucketAlreadyOwnedByYou: message: Your previous request to create the named bucket succeeded and you already own it.\n\nDuring handling of the above exception, another exception occurred:\n\nTraceback (most recent call last):\n  File \"/mint/run/core/minio-py/tests.py\", line 1968, in main\n    test_make_bucket_default_region(client, log_output)\n  File \"/mint/run/core/minio-py/tests.py\", line 196, in test_make_bucket_default_region\n    raise Exception(err)\nException: BucketAlreadyOwnedByYou: message: Your previous request to create the named bucket succeeded and you already own it.\n",
  "status": "FAIL"
}
(5/15) Running s3cmd tests ... done in 44 seconds
(6/15) Running s3select tests ... FAILED in 22 minutes and 14 seconds
{
  "name": "s3select:test_csv_input_quote_char",
  "function": "select_object_content(bucket_name, object_name, opts)",
  "args": {
    "bucket_name": "s3select-test-d726e11f-829d-42c2-8b3f-e4b2e07951c7"
  },
  "duration": 1333961,
  "message": "BucketNotEmpty: message: The bucket you tried to delete is not empty.",
  "error": "Traceback (most recent call last):\n  File \"./tests.py\", line 45, in main\n    test_csv_input_custom_quote_char(client, log_output)\n  File \"/mint/run/core/s3select/csv.py\", line 110, in test_csv_input_custom_quote_char\n    client.remove_bucket(bucket_name)\n  File \"/usr/local/lib/python3.6/dist-packages/minio/api.py\", line 446, in remove_bucket\n    self._url_open('DELETE', bucket_name=bucket_name)\n  File \"/usr/local/lib/python3.6/dist-packages/minio/api.py\", line 2020, in _url_open\n    object_name).get_exception()\nminio.error.BucketNotEmpty: BucketNotEmpty: message: The bucket you tried to delete is not empty.\n",
  "status": "FAIL"
}
(6/15) Running security tests ... done in 0 seconds

Executed 6 out of 15 tests successfully.
```
<a name="3b70e2a0-a669-46fd-be13-c8f4a487ad1d"></a>
### 9917-3821458/mint-xl.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-dev3.minio.io:31346
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 176ddf710022:/mint/log /tmp/mint-logs'

(1/15) Running aws-sdk-go tests ... done in 1 seconds
(2/15) Running aws-sdk-java tests ... done in 3 seconds
(3/15) Running aws-sdk-php tests ... done in 42 seconds
(4/15) Running aws-sdk-ruby tests ... done in 3 seconds
(5/15) Running awscli tests ... done in 1 minutes and 18 seconds
(6/15) Running healthcheck tests ... done in 1 seconds
(7/15) Running mc tests ... done in 24 seconds
(8/15) Running minio-dotnet tests ... done in 42 seconds
(9/15) Running minio-go tests ... FAILED in 1 minutes and 4 seconds
{
  "args": {
    "bucketName": "minio-go-test-kg0s3ta6maythaoi",
    "objectName": "bvg3rj3bipgeqrt0j4lx2e62trzyli",
    "objectPrefix": "",
    "recursive": "true"
  },
  "duration": 3342,
  "error": "The bucket you tried to delete is not empty. You must delete all versions in the bucket.",
  "function": "RemoveObjectsWithOptions(bucketName, objectsCh, opts)",
  "message": "Cleanup failed",
  "name": "minio-go: testRemoveObjectsWithOptions",
  "status": "FAIL"
}
(9/15) Running minio-java tests ... FAILED in 2 minutes and 17 seconds
{
  "name": "minio-java",
  "function": "enableObjectLegalHold(String bucketName, String objectName, String versionId)",
  "duration": 283,
  "status": "FAIL",
  "error": "error occurred\nErrorResponse(code = BucketNotEmpty, message = The bucket you tried to delete is not empty. You must delete all versions in the bucket., bucketName = minio-java-test-17tqfuu, objectName = null, resource = /minio-java-test-17tqfuu, requestId = 161BDAE950708BE3, hostId = 2843e25e-0271-4f79-a1af-9815a6c8b5d0)\nrequest={method=DELETE, url=http://minio-dev3.minio.io:31346/minio-java-test-17tqfuu, headers=Host: minio-dev3.minio.io:31346\nAccept-Encoding: identity\nUser-Agent: MinIO (amd64; amd64) minio-java/dev\nContent-MD5: 1B2M2Y8AsgTpgAmY7PhCfg==\nx-amz-content-sha256: e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855\nx-amz-date: 20200625T175308Z\nAuthorization: AWS4-HMAC-SHA256 Credential=*REDACTED*/20200625/us-east-1/s3/aws4_request, SignedHeaders=accept-encoding;content-md5;host;x-amz-content-sha256;x-amz-date, Signature=*REDACTED*\n}\nresponse={code=409, headers=Accept-Ranges: bytes\nContent-Length: 373\nContent-Security-Policy: block-all-mixed-content\nContent-Type: application/xml\nServer: MinIO/DEVELOPMENT.2020-06-25T17-40-36Z\nVary: Origin\nX-Amz-Request-Id: 161BDAE950708BE3\nX-Xss-Protection: 1; mode=block\nDate: Thu, 25 Jun 2020 17:53:08 GMT\n}\n >>> [io.minio.MinioClient.execute(MinioClient.java:1211), io.minio.MinioClient.execute(MinioClient.java:1236), io.minio.MinioClient.executeDelete(MinioClient.java:1366), io.minio.MinioClient.removeBucket(MinioClient.java:3686), FunctionalTest.enableObjectLegalHold_test(FunctionalTest.java:3314), FunctionalTest.runTests(FunctionalTest.java:4079), FunctionalTest.main(FunctionalTest.java:4192)]"
}
(9/15) Running minio-js tests ... done in 50 seconds
(10/15) Running minio-py tests ... done in 2 minutes and 30 seconds
(11/15) Running s3cmd tests ... done in 18 seconds
(12/15) Running s3select tests ... done in 4 seconds
(13/15) Running security tests ... done in 0 seconds

Executed 13 out of 15 tests successfully.
```
<a name="5a3ffdf7-6c89-4edd-b4f9-c3a44c49df72"></a>
### 9917-3821458/mint-dist-xl.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-dev6.minio.io:32219
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 2137a975b734:/mint/log /tmp/mint-logs'

(1/15) Running aws-sdk-go tests ... FAILED in 6 minutes and 12 seconds
{
  "alert": "",
  "args": {
    "bucketName": "aws-sdk-go-test-3t3q19qhfvmb0u",
    "expiry": 60000000000,
    "objectName1": "testObject1",
    "objectName2": "testObject2"
  },
  "duration": 247139,
  "error": "BucketNotEmpty: The bucket you tried to delete is not empty\n\tstatus code: 409, request id: 161BDAE3CF50C04C, host id: ",
  "function": "testListObjects",
  "message": "AWS SDK Go DeleteBucket Failed",
  "name": "aws-sdk-go",
  "status": "FAIL"
}
(1/15) Running aws-sdk-java tests ... done in 3 seconds
(2/15) Running aws-sdk-php tests ... FAILED in 8 seconds
{
  "name": "aws-sdk-php",
  "function": "copyObject ( array $params = [] )",
  "args": {
    "Bucket": "aws-sdk-php-5ef4e470d3aa1",
    "Object": "obj1"
  },
  "duration": "818",
  "status": "FAIL",
  "error": "Error executing \"CopyObject\" on \"http://minio-dev6.minio.io:32219/aws-sdk-php-5ef4e470d3aa1/obj1-copy\"; AWS HTTP error: Server error: `PUT http://minio-dev6.minio.io:32219/aws-sdk-php-5ef4e470d3aa1/obj1-copy` resulted in a `503 Service Unavailable` response:\n<?xml version=\"1.0\" encoding=\"UTF-8\"?>\n<Error><Code>SlowDown</Code><Message>Please reduce your request</Message><Key>obj (truncated...)\n SlowDown (server): Please reduce your request - <?xml version=\"1.0\" encoding=\"UTF-8\"?>\n<Error><Code>SlowDown</Code><Message>Please reduce your request</Message><Key>obj1-copy</Key><BucketName>aws-sdk-php-5ef4e470d3aa1</BucketName><Resource>/aws-sdk-php-5ef4e470d3aa1/obj1-copy</Resource><RequestId>161BDAE68347F3A7</RequestId><HostId>f4e2a3f8-fe83-479d-aeac-363c8354ec2e</HostId></Error>"
}
(2/15) Running aws-sdk-ruby tests ... FAILED in 8 seconds
{
  "name": "aws-sdk-ruby",
  "function": "removeBucket(bucket_name)",
  "args": {
    "bucket_name": "aws-sdk-ruby-bucket-0ab9e16c5562"
  },
  "duration": 3221.16,
  "error": "Failed to clean-up bucket 'aws-sdk-ruby-bucket-0ab9e16c5562', The bucket you tried to delete is not empty",
  "status": "FAIL"
}
(2/15) Running awscli tests ... FAILED in 14 seconds
{
  "name": "awscli",
  "duration": 8187,
  "function": "delete_bucket\n",
  "status": "FAIL",
  "error": "delete: s3://awscli-mint-test-bucket-1185/datafile-1-kBremove_bucket failed: s3://awscli-mint-test-bucket-1185 An error occurred (BucketNotEmpty) when calling the DeleteBucket operation: The bucket you tried to delete is not empty"
}
(2/15) Running healthcheck tests ... done in 0 seconds
(3/15) Running mc tests ... done in 51 seconds
(4/15) Running minio-dotnet tests ... done in 1 minutes and 9 seconds
(5/15) Running minio-go tests ... FAILED in 2 seconds
{
  "args": {
    "bucketName": "minio-go-test-mlmrwwj6scy4se0m",
    "objectName": "q31eumc94f2gfhqpwfqepjjogupw46",
    "region": "eu-west-1"
  },
  "duration": 2735,
  "error": "The bucket you tried to delete is not empty",
  "function": "MakeBucket(bucketName, region)",
  "message": "Cleanup failed",
  "name": "minio-go: testGetObjectClosedTwiceV2",
  "status": "FAIL"
}
(5/15) Running minio-java tests ... FAILED in 2 seconds
{
  "name": "minio-java",
  "function": "listBuckets()",
  "duration": 29,
  "status": "FAIL",
  "error": "error occurred\nErrorResponse(code = SlowDown, message = Please reduce your request, bucketName = minio-java-test-2vbpu8k, objectName = null, resource = /minio-java-test-2vbpu8k, requestId = 161BDB0878D0466B, hostId = f4e2a3f8-fe83-479d-aeac-363c8354ec2e)\nrequest={method=PUT, url=http://minio-dev6.minio.io:32219/minio-java-test-2vbpu8k, headers=Host: minio-dev6.minio.io:32219\nAccept-Encoding: identity\nUser-Agent: MinIO (amd64; amd64) minio-java/dev\nContent-MD5: 1B2M2Y8AsgTpgAmY7PhCfg==\nx-amz-content-sha256: e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855\nx-amz-date: 20200625T175522Z\nAuthorization: AWS4-HMAC-SHA256 Credential=*REDACTED*/20200625/us-east-1/s3/aws4_request, SignedHeaders=accept-encoding;content-md5;host;x-amz-content-sha256;x-amz-date, Signature=*REDACTED*\n}\nresponse={code=503, headers=Accept-Ranges: bytes\nContent-Length: 305\nContent-Security-Policy: block-all-mixed-content\nContent-Type: application/xml\nRetry-After: 120\nServer: MinIO/DEVELOPMENT.2020-06-25T17-40-36Z\nVary: Origin\nX-Amz-Request-Id: 161BDB0878D0466B\nX-Xss-Protection: 1; mode=block\nDate: Thu, 25 Jun 2020 17:55:22 GMT\n}\n >>> [io.minio.MinioClient.execute(MinioClient.java:1211), io.minio.MinioClient.execute(MinioClient.java:1236), io.minio.MinioClient.executePut(MinioClient.java:1410), io.minio.MinioClient.makeBucket(MinioClient.java:3286), io.minio.MinioClient.makeBucket(MinioClient.java:3199), FunctionalTest.listBuckets_test(FunctionalTest.java:324), FunctionalTest.runTests(FunctionalTest.java:4011), FunctionalTest.main(FunctionalTest.java:4192)]"
}
(5/15) Running minio-js tests ... FAILED in 45 seconds
{
  "name": "minio-js",
  "function": "\"after all\" hook in \"functional tests\"",
  "duration": 8,
  "status": "FAIL",
  "error": "S3Error: Please reduce your request at Object.parseError (node_modules/minio/dist/main/xml-parsers.js:86:11) at /mint/run/core/minio-js/node_modules/minio/dist/main/transformers.js:156:22 at DestroyableTransform._flush (node_modules/minio/dist/main/transformers.js:80:10) at DestroyableTransform.prefinish (node_modules/readable-stream/lib/_stream_transform.js:129:10) at prefinish (node_modules/readable-stream/lib/_stream_writable.js:611:14) at finishMaybe (node_modules/readable-stream/lib/_stream_writable.js:620:5) at endWritable (node_modules/readable-stream/lib/_stream_writable.js:643:3) at DestroyableTransform.Writable.end (node_modules/readable-stream/lib/_stream_writable.js:571:22) at IncomingMessage.onend (_stream_readable.js:673:10) at endReadableNT (_stream_readable.js:1201:12) at processTicksAndRejections (internal/process/task_queues.js:84:21)"
}
(5/15) Running minio-py tests ... FAILED in 6 minutes and 5 seconds
{
  "name": "minio-py:test_fput_object_small_file",
  "function": "fput_object(bucket_name, object_name, file_path, content_type, metadata, sse, progress, part_size)",
  "args": {
    "bucket_name": "minio-py-test-7ccd2496-6dcd-41e7-9f9e-7f32ec3cd5b6",
    "object_name": "238f0308-fc83-481b-adc7-2e928318c836",
    "file_path": "/mint/data/datafile-1-MB",
    "metadata": {
      "x-amz-storage-class": "STANDARD_IA"
    }
  },
  "duration": 240626,
  "message": "BucketNotEmpty: message: The bucket you tried to delete is not empty.",
  "error": "Traceback (most recent call last):\n  File \"/mint/run/core/minio-py/tests.py\", line 371, in test_fput_object_small_file\n    client.remove_bucket(bucket_name)\n  File \"/usr/local/lib/python3.6/dist-packages/minio/api.py\", line 446, in remove_bucket\n    self._url_open('DELETE', bucket_name=bucket_name)\n  File \"/usr/local/lib/python3.6/dist-packages/minio/api.py\", line 2020, in _url_open\n    object_name).get_exception()\nminio.error.BucketNotEmpty: BucketNotEmpty: message: The bucket you tried to delete is not empty.\n\nDuring handling of the above exception, another exception occurred:\n\nTraceback (most recent call last):\n  File \"/mint/run/core/minio-py/tests.py\", line 1983, in main\n    test_fput_object_small_file(client, testfile, log_output)\n  File \"/mint/run/core/minio-py/tests.py\", line 373, in test_fput_object_small_file\n    raise Exception(err)\nException: BucketNotEmpty: message: The bucket you tried to delete is not empty.\n",
  "status": "FAIL"
}
(5/15) Running s3cmd tests ... done in 29 seconds
(6/15) Running s3select tests ... FAILED in 38 minutes and 5 seconds
{
  "name": "s3select:test_csv_input_quote_char",
  "function": "select_object_content(bucket_name, object_name, opts)",
  "args": {
    "bucket_name": "s3select-test-d34cbd31-91c8-4b22-a6c7-7366f5bf1aa2"
  },
  "duration": 2285073,
  "message": "BucketNotEmpty: message: The bucket you tried to delete is not empty.",
  "error": "Traceback (most recent call last):\n  File \"./tests.py\", line 45, in main\n    test_csv_input_custom_quote_char(client, log_output)\n  File \"/mint/run/core/s3select/csv.py\", line 110, in test_csv_input_custom_quote_char\n    client.remove_bucket(bucket_name)\n  File \"/usr/local/lib/python3.6/dist-packages/minio/api.py\", line 446, in remove_bucket\n    self._url_open('DELETE', bucket_name=bucket_name)\n  File \"/usr/local/lib/python3.6/dist-packages/minio/api.py\", line 2020, in _url_open\n    object_name).get_exception()\nminio.error.BucketNotEmpty: BucketNotEmpty: message: The bucket you tried to delete is not empty.\n",
  "status": "FAIL"
}
(6/15) Running security tests ... done in 0 seconds

Executed 6 out of 15 tests successfully.
```
Deleting image on docker hub
Deleting image locally

--- a/cmd/peer-rest-server.go
+++ b/cmd/peer-rest-server.go
@@ -606,6 +606,10 @@ func (s *peerRESTServer) LoadBucketMetadataHandler(w http.ResponseWriter, r *htt
 	}
 
 	globalBucketMetadataSys.Set(bucketName, meta)
+
+	if meta.notificationConfig != nil {
+		globalNotificationSys.AddRulesMap(bucketName, meta.notificationConfig.ToRulesMap())
+	}
 }
 
 // ReloadFormatHandler - Reload Format.

